examples/hosts/app-integration/external-data/README.md

```diff
@@ -6,15 +6,62 @@ Please note that the ideas explored here are experimental and under development.
+3. Sync updates between Fluid and the source of truth in as close to real-time as the scenario allows.
+
+In this case, the Fluid collaboration session serves as a "drafting surface" in which clients collaborate to create a draft of the data and then send the saved data back to the source of truth for long term storage.
 
 ## Strategy overview
 
-TBD
+In order to accomplish the goals above, we have split up the responsibilities into a few different pieces:
+
+**External Data Service**
+
+Many services that would hold the "source of truth" data offer explicit commit style interfaces (e.g. vi REST call or similar) which are not well suited to rapid updates.
+However, they often expose third-party integration via REST APIS for querying and manipulating data, as well as webhooks for watching updates to the data.
+
+This repo contains a service that mocks the external "source of truth" data server. This mock service offers a REST API collection and webhook interfaces in `./src/mock-external-data-service`. The API requests served by this "external" service are the following:
+
+1. POST `/register-for-webhook`: Register's the sender's URL to receive notifications when the external task-list data changes. Currently, the Customer Service registers its `/echo-external-data-webhook` endpoint here to be called when data changes.
+2. GET `/fetch-tasks`: Fetches the task list from the external data store. Called by the Fluid client.
+3. POST `/set-tasks`: Updates external data store with new tasks list (complete override). Called by the Fluid client.
+4. POST `/debug-reset-task-list`: Resets the external data to its original contents. Called by the Fluid client.
+
+Find the details of the API in the [External Data Service README](./src/mock-external-data-service/README.md)
+
+**Customer Service**
+
+Next we need a customer service that functions as the intermediary between the External Data Service and the Fluid Service. This server is responsible for authenticating to the external service on the customer's behalf. It registers to the External Data Service webhooks and listens for incoming changes. It also acts as a translation layer, translating to and from the External Data Service expected format and the Fluid Service's expected format.
+
+In this example, the Customer Service contains the following endpoints:
+
+1. POST `/register-for-webhook`: Registers the sender's URL to receive notifications when the external task-list data changes. Currently, Alfred registers it's `/task-list-hook` endpoint to this URL to be called when the Customer Service receives a notification on its `/echo-external-data-webhook` endpoint.
+2. POST `/echo-external-data-webhook`: "Echoes" the external data services data update notifications to our own webhook subscribers. This data will be forwarded to our own subscribers. Registered to be called by the External Data Service when there's been a change to data upstream.
```

Comment:
I still think we should remove this from the documentation here. It should be considered a private implementation detail of the customer service.

Reply:
Thinking about this more, I don't think we should remove it because this example app is to prototype the full flow. It's role is to provide a skeleton of how a customer would eventually implement this. Maybe I can add text here to clarify that, but removing it would leave the prototype incomplete. Interested in @ChumpChief's thoughts on whether we should remove this endpoint from the docs?

Reply:
Maybe drop the "echo" info (yes we'll use it to echo internally, but the endpoint doesn't need to reflect that externally) and make it something like

```suggestion
2. POST `/external-data-webhook`: Registered to be called by the External Data Service webhook when there's been a change to data upstream.
```

And then the "echo webhook" section of the documentation can cover the internal behavior of echoing the webhook.

Reply:
I guess I'm fine with leaving docs about it if we want to describe the full flow, but we should note that this is a route established by the customer service to be used exclusively as a subscription endpoint for the external data service's webhook, and should be considered a private implementation detail to the service.